woocommerce/woocommerce-ios
# Support order creation notes

# Why 
Fix issue #6386 by defining a customer note section inside the New Order view capable of creating, editing or removing any text as the Order customer note field.

# How

Defines the `CustomerOrderNoteSection` inside the `NewOrder` view and handles all user inputs inside the `NewOrderViewModel` taking advantage of the already existing `EditCustomerNote` view.

| New Order view with the empty Customer note section  | Editing the customer note | New Order view with an existing Customer note input | Order created with a customer note |
| ------------- | ------------- | ------------- | ------------- |
| ![Simulator Screen Shot - iPhone 13 Pro Max - 2022-03-16 at 19 07 58](https://user-images.githubusercontent.com/5920403/158699477-7e075719-da78-449d-82d3-629404fa0047.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2022-03-16 at 19 08 06](https://user-images.githubusercontent.com/5920403/158699488-3842fe81-5da2-45e8-8a42-96bbb7f57846.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2022-03-16 at 19 08 08](https://user-images.githubusercontent.com/5920403/158699503-656e02c1-3f54-4d49-ac65-e7ef7a817f69.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2022-03-16 at 19 08 19](https://user-images.githubusercontent.com/5920403/158699512-146ddf1f-c5e0-4991-9a83-e60ae1d7a38e.png) |

# Testing Steps

- Make sure the Order Creation experimental toggle is active inside the app settings
- Go to the Order list, hit the `+` icon and select `Order Creation`
- Verify the `Customer note` section, interact with it by adding, editing, or removing notes from it
- Make sure the discard warning works as expected with the customer note presence or not

 
Update release notes:
